packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -2097,6 +2103,7 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
         removeTopPadding: true,
         removeRightPadding: false,
         removeBottomPadding: false,
+        maintainBottomViewPadding: !_resizeToAvoidBottomInset,
```

Comment:
Is the navigation bar really the only place affected?

Reply:
I was only able to reproduce the bug in this case. After chatting offline I'm going to check:

- SnackBar
- PersistentFooterButtons
- BottomSheet
- Floating Action Button

once more and write tests to include them.